src/Interfaces/AbstractHyperrectangle.jl

```diff
@@ -56,8 +56,20 @@ Return the generator matrix of a hyperrectangular set.
 
 A matrix where each column represents one generator of `H`.
 """
-function genmat(H::AbstractHyperrectangle)
-    return genmat_fallback(H)
+function genmat(H::AbstractHyperrectangle{N}) where {N<:Real}
+    r = radius_hyperrectangle(H)
+    if any(iszero, r)
```

Comment:
I would return a `Diagonal(sparse(r))` in general here. Other code can take advantage knowning that this matrix is `Diagonal`. And returning two different matrix types may introduce a type-instability.

Reply:
This was my first implementation, but then a test fails. We assume that the generator matrix has no zero columns.

Reply:
Do you remember which test fails? I don't see any reason why `Diagonal(r)` would be wrong; it probably uncovers another bug..?

Reply:
https://github.com/JuliaReach/LazySets.jl/blob/fd488b412a0643e94f6e18a800b02bbaf3c2aaa3/test/unit_Hyperrectangle.jl#L26-L30

Reply:
I changed it to always returning a `SparseMatrix`. Problem solved :+1: